library/swift/test/StreamClientBuilderTests.swift

```diff
@@ -172,6 +172,7 @@ final class StreamClientBuilderTests: XCTestCase {
                                     dnsRefreshSeconds: 300,
                                     dnsFailureRefreshSecondsBase: 400,
                                     dnsFailureRefreshSecondsMax: 500,
+                                    filterChain: [],
```

Comment:
We should add tests for this new functionality

Reply:
Agreed. To some degree the example app acts as an end-to-end integration test (and I actually plan to update it with n explicit check on the filter's effect). But I can look into what level of testing we can do at the unit layer here as well.